test/galois/Main.hs

```diff
@@ -0,0 +1,235 @@
+module Main (main) where
+
+import Control.Monad (guard)
+import GHC.IO.Encoding (setLocaleEncoding, utf8)
+import Grumplestiltskin.Galois (
+    PGFElement,
+    pgfExp,
+    pgfFromElem,
+    pgfFromInteger,
+    pgfFromPInteger,
+    pgfRecip,
+    pgfToElem,
+ )
+import Numeric.Natural (Natural)
+import Plutarch.Prelude (
+    PBool,
+    PInteger,
+    PNatural,
+    PPositive,
+    S,
+    Term,
+    pconstant,
+    plam,
+    plift,
+    pnegate,
+    ppowNatural,
+    ppowPositive,
+    ptryNatural,
+    pupcast,
+    (#),
+    (#*),
+    (#+),
+    (#-),
+    (#==),
+ )
+import Plutarch.Test.Utils (precompileTerm)
+import Test.QuickCheck (
+    Gen,
+    Property,
+    arbitrary,
+    chooseInt,
+    forAll,
+    forAllShrink,
+    shrink,
+ )
+import Test.QuickCheck.Instances.Natural ()
+import Test.Tasty (adjustOption, defaultMain, testGroup)
+import Test.Tasty.QuickCheck (QuickCheckTests, testProperty)
+
+main :: IO ()
+main = do
+    -- Pre-emptively avoid locale encoding issues
+    setLocaleEncoding utf8
+    defaultMain . adjustOption moreTests . testGroup "Tests" $
+        [ testGroup
+            "PGFIntermediate"
+            [ testProperty "#+ commutes" propCommAdd
+            , testProperty "#+ associates" propAssocAdd
+            , testProperty "pnegate produces an additive inverse" propNegate
+            , testProperty "#* commutes" propCommMul
+            , testProperty "#* associates" propAssocMul
+            , testProperty "pgfRecip produces a reciprocal" propRecip
+            , testProperty "distributivity of #+ over #*" propDistribute
+            , testProperty "pgfExp equivalent to ppowPositive" propExpPos
+            , testProperty "pgfExp equivalent to ppowNatural" propExpNat
+            , testProperty "x ^ n * x ^ -n = 1 for nonzero x" propExpRing
```

Comment:
I think there should be additional properties explicitly tested, i.e., the identity properties. Additive identity `a + 0 = a`, and multiplicative identity `a * 1 = a`...

Otherwise LGTM